File: ProblemSets/pset1(2)/cash.c
#include <cs50.h>
#include <stdio.h>
int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);
int main(void)
{
    int cents = get_cents();
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;
    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;
    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;
    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;
    int coins = quarters + dimes + nickels + pennies;
    printf("%i\n", coins);
}
int get_cents(void)
{
    int n;
    do
    {
        n = get_int("Number of cents: ");
    } while (n < 1);
    return n;
}
int calculate_quarters(int cents)
{
    int quarters = cents / 25;
    return quarters;
}
int calculate_dimes(int cents)
{
    int dimes = cents / 10;
    return dimes;
}
int calculate_nickels(int cents)
{
    int nickels = cents / 5;
    return nickels;
}
int calculate_pennies(int cents)
{
    int pennies = cents / 1;
    return pennies;
}

File: ProblemSets/pset6(2)/sentimental-cash/cash.py
from cs50 import get_float

def get_cents():
    while True:
        n = get_float("Change owed: ")
        if n >= 0:
            break
        else:
            print("Please enter a non-negative value.")
    return round(n * 100)

def calculate_coins(cents, denomination):
    coins = cents // denomination
    cents = cents % denomination
    return coins, cents

def main():
    cents = get_cents()

    quarters, cents = calculate_coins(cents, 25)
    dimes, cents = calculate_coins(cents, 10)
    nickels, cents = calculate_coins(cents, 5)
    pennies = cents

    total_coins = quarters + dimes + nickels + pennies
    print(total_coins)


main()
